src/core/database.h

```diff
@@ -196,7 +200,7 @@ public :
 																																														"RECONSTRUCTION_ID", "SHOULD_AUTOMASK", "SHOULD_REFINE_INPUT_PARAMS", "SHOULD_USE_SUPPLIED_MASK",
 																																														"MASK_ASSET_ID", "MASK_EDGE_WIDTH", "OUTSIDE_MASK_WEIGHT", "SHOULD_LOWPASS_OUTSIDE_MASK", "MASK_FILTER_RESOLUTION");};
 
-
+	bool CreateMovieMetadataTable() { return CreateTable("MOVIE_ASSETS_METADATA", "Ptt", "MOVIE_ASSET_ID", "METADATA_TYPE", "JSON"); }
```

Comment:
- metadatatype should store a string that indicates the type of metadata. For now the only type is SerialEM FrameSet mdocs, but in the future it could include EPU XML files or SerialEM tilt-series mdocs. 
- JSON holds the actual metadata in JSON format. An alternative approach would be to create a column for every field of the metadata, but that would be less flexible if new fields are included in the future and for accommodation metadata from different programs.

Reply:
> 
> 
>     * metadatatype should store a string that indicates the type of metadata. For now the only type is SerialEM FrameSet mdocs, but in the future it could include EPU XML files or SerialEM tilt-series mdocs.

Okay, then maybe something like **METADATA_SOURCE_SOFTWARE** or **METADATA_FORMAT**

> 
>     * JSON holds the actual metadata in JSON format. An alternative approach would be to create a column for every field of the metadata, but that would be less flexible if new fields are included in the future and for accommodation metadata from different programs.

Oh cool! I like the flexibility that provides. How will you deal with metadata that is/isn't present? The database will just store and return whatever is there, so it won't break (which is rad) so will it just be program specific error handling?

Reply:
Also with the name, perhaps **JSON_TABLE** or something to indicate what it is given that JSON itself is a data storage *format*

Reply:
So I think if you plan to support other file types in the future, you may want to have the important points as database columns.  Otherwise each program will have to check the format and read things in a different way depending on that etc.  Adding columns later isn't necessarily a problem.  Why don't you keep the whole JSON string, but also specifically fill columns for the data you think you will need in the near future?